src/api/user/UserAPISpec.js

```diff
@@ -0,0 +1,117 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2021, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+import {
+    createOpenMct,
+    resetApplicationState
+} from '../../utils/testing';
+import {
+    MULTIPLE_PROVIDER_ERROR
+} from './constants';
+import ExampleUserProvider from '../../plugins/exampleUser/ExampleUserProvider';
+
+const USERNAME = 'Test User';
+const EXAMPLE_ROLE = 'example-role';
+
+describe("The User API", () => {
+    let openmct;
+
+    beforeEach(() => {
+        openmct = createOpenMct();
+    });
+
+    afterEach(() => {
+        return resetApplicationState(openmct);
+    });
+
+    describe('with regard to user providers', () => {
+
+        it('allows you to specify a user provider', () => {
+            openmct.user.setProvider(new ExampleUserProvider(openmct));
+
+            expect(openmct.user._provider).toBeInstanceOf(ExampleUserProvider);
```

Comment:
can we test this without knowing about internals?

It looks like what you've done in the example spec is what should be done here. Testing that providers don't exist, and then they do when added. And if the provider needs to be checked, the check example spec has of `isInstanceOf` should do the trick.